src/rgw/driver/rados/rgw_putobj_processor.cc

```diff
@@ -341,6 +341,34 @@ int AtomicObjectProcessor::complete(size_t accounted_size,
   obj_op.meta.zones_trace = zones_trace;
   obj_op.meta.modify_tail = true;
 
+  map<string, bufferlist>::iterator attr_iter = attrs.find(RGW_ATTR_CLOUD_TIER_TYPE);
+  if (attr_iter != attrs.end()) {
+    string m;
+    auto i = attr_iter->second.cbegin();
+    try {
+      using ceph::decode;
+      decode(m, i);
+    } catch (buffer::error& err) {
+    }
+    if (m == "cloud-s3") {
+      obj_op.meta.category = RGWObjCategory::CloudTiered;
+      manifest.set_tier_type("cloud-s3");
+      map<string, bufferlist>::iterator config_iter = attrs.find(RGW_ATTR_CLOUD_TIER_CONFIG);
+      if (config_iter != attrs.end()) {
+        RGWObjTier tier_config;
+        auto i = config_iter->second.cbegin();
+        try {
+          using ceph::decode;
+          decode(tier_config, i);
+          manifest.set_tier_config(tier_config);
+          attrs.erase(RGW_ATTR_CLOUD_TIER_CONFIG);
+        } catch (buffer::error& err) {
+        }
+      }
+    }
+    attrs.erase(RGW_ATTR_CLOUD_TIER_TYPE);
```

Comment:
`attr_iter` is an iterator to this element. you can `attrs.erase(attr_iter)` to avoid searching for it again. same for `config_iter` and `attrs.erase(RGW_ATTR_CLOUD_TIER_CONFIG)` above